### PR TITLE
Pusher integration into live monitoring.

### PR DESCRIPTION
**Pull request by HelenaEksler**

### CODE REVIEW

File: client/app/scripts/services/repos.js [hunk 1]
```diff
@@ -48,7 +48,7 @@ angular.module('clientApp')
      */
     function getDataFromBackend(repoId) {
       var deferred = $q.defer();
-      var url = Config.backend + '/api/github_repos';
```
Comment (amitaibu) [inline client/app/scripts/services/repos.js:51]: this cannot change - it what gets the user repos from github. you can create another service.

Comment (HelenaEksler) [inline client/app/scripts/services/repos.js:51]: Sorry, I've forgot to change it back. I've set to repositories to test.

File: shoov/modules/custom/shoov_ci_build/shoov_ci_build.module [hunk 2]
```diff
@@ -177,6 +177,9 @@ function shoov_ci_build_create_build_item($node, $timestamp = NULL) {
   $account = user_load($node->uid);
   flag('flag', 'subscribe_ci_builds', $node->nid, $account);
 
+  // Trigger Pusher event when new CI Build is created.
+  shoov_ci_build_trigger_event(message_load($message->mid), 'new_ci_build');
```
Comment (amitaibu) [inline shoov/modules/custom/shoov_ci_build/shoov_ci_build.module]: `ci_build_new`

File: shoov/modules/custom/shoov_ci_build/shoov_ci_build.module [hunk 3]
```diff
@@ -204,6 +207,9 @@ function shoov_ci_build_message_update(Message $message) {
 
   shoov_ci_build_create_build_item($node, $timestamp);
   shoov_ci_build_process_ci_build_item($message);
+
+  // Trigger Pusher event when CI Build status changes.
+  shoov_ci_build_trigger_event($message, 'update_ci_build');
```
Comment (amitaibu) [inline shoov/modules/custom/shoov_ci_build/shoov_ci_build.module]: `ci_build_update`

File: client/app/scripts/controllers/repo.js [hunk 4]
```diff
@@ -8,10 +8,35 @@
...
-  .controller('RepoCtrl', function ($scope, build, ciBuildItems, incidents) {
+  .controller('RepoCtrl', function ($scope, build, ciBuildItems, incidents, channelManager) {
     $scope.build = build[0];
     $scope.ciBuildItems = ciBuildItems;
     $scope.incidents = incidents;
 
     $scope.parseInt = parseInt;
+
+    var channels = channelManager.getChannels();
+    console.log(channels);
```
Comment (amitaibu) [inline client/app/scripts/controllers/repo.js]: remove

File: client/app/scripts/controllers/repo.js [hunk 5]
```diff
@@ -8,10 +8,35 @@
...
+    angular.forEach(channels, function(channel) {
+      channel.bind('new_incident', function(data) {
```
Comment (amitaibu) [inline client/app/scripts/controllers/repo.js]: `ci_incident_new`

File: client/app/scripts/controllers/repo.js [hunk 6]
```diff
@@ -8,10 +8,35 @@
...
+        // Put new item in the begginning of the list.
+        $scope.ciBuildItems.unshift(data[0]);
+      });
+
+      channel.bind('update_ci_build', function(data) {
+        var id = data[0].id;
+        // Update the existing item.
+        angular.forEach($scope.ciBuildItems, function(item, itemId) {
+          if (item.id != id) {
+            return;
```
Comment (amitaibu) [inline client/app/scripts/controllers/repo.js]: `// Item already exists in the list.`

Comment (amitaibu) [inline client/app/scripts/controllers/repo.js:37]: comment

File: shoov/modules/custom/shoov_ci_build/shoov_ci_build.module [hunk 7]
```diff
@@ -387,3 +393,27 @@ function shoov_ci_build_get_nodejs_server_url(Message $message) {
...
+ */
+function shoov_ci_build_trigger_event(Message $message, $event) {
+  $wrapper = entity_metadata_wrapper('message', $message);
+
+  $account = user_load($message->uid);
+  $handler = restful_get_restful_handler('ci-build-items');
+  $handler->setAccount($account);
+  $result = $handler->get($message->mid);
+  $data = array($result[0]);
+  if (isset($wrapper->field_ci_build->og_repo)) {
```
Comment (amitaibu) [inline shoov/modules/custom/shoov_ci_build/shoov_ci_build.module]: This is not clear.

File: shoov/modules/custom/shoov_ci_build/shoov_ci_build.module [hunk 8]
```diff
@@ -387,3 +393,27 @@ function shoov_ci_build_get_nodejs_server_url(Message $message) {
...
+    shoov_pusher_trigger_event($wrapper->field_ci_build->og_repo->getIdentifier(), $event, $data);
+  }
+  else {
+    shoov_pusher_trigger_user_event($message->uid, $event, $data);
```
Comment (amitaibu) [inline shoov/modules/custom/shoov_ci_build/shoov_ci_build.module]: remove

File: shoov/modules/custom/shoov_ci_incident/shoov_ci_incident.module [hunk 9]
```diff
@@ -80,6 +80,15 @@ function shoov_ci_incident_create_error_incident(Message $message) {
...
   $wrapper->save();
 
+  $account = user_load($node->uid);
+  $handler = restful_get_restful_handler('ci-incidents');
+  $handler->setAccount($account);
+  $result = $handler->get($node->nid);
+  $data = array($result[0]);
+
+  $repo_id = $wrapper->og_repo->value(array('identifier' => TRUE));
+  shoov_pusher_trigger_event($repo_id, 'new_incident', $data);
```
Comment (amitaibu) [inline shoov/modules/custom/shoov_ci_incident/shoov_ci_incident.module]: `ci_incident_new`

**Comment by HelenaEksler** [comment]
shoov7

**Comment by amitaibu** [comment]
Merged, thanks.

**Pull request closed without merging**